app/components/TablePaginationControls.tsx

```diff
@@ -31,6 +30,10 @@ export default function TablePaginationControls({
   const currentPageNum = queryConfig.offset / queryConfig.limit + 1;
   const totalRecords = aggregateData?.aggregate?.count || 0;
 
+  const pageLeftButtonDisabled =
+    recordCount === 0 || queryConfig.offset === 0 || isLoading;
+  const pageRightButtonDisabled = recordCount < queryConfig.limit || isLoading;
```

Comment:
This change, although it may seem small, transmits a lot of information to the user in a subtle way. Thanks so much for adding this.